pkg/webhooks/networkpolicies/networkpolicies.go

```diff
@@ -0,0 +1,195 @@
+}
+
+// NewWebhook creates the new webhook
+func NewWebhook() *networkpoliciesruleWebhook {
+	scheme := runtime.NewScheme()
+	return &networkpoliciesruleWebhook{
+		s: *scheme,
+	}
+}
+
+// Authorized implements Webhook interface
+func (s *networkpoliciesruleWebhook) Authorized(request admissionctl.Request) admissionctl.Response {
+	//Implement authorized next
+	return s.authorized(request)
+}
+
+func (s *networkpoliciesruleWebhook) authorized(request admissionctl.Request) admissionctl.Response {
+	var ret admissionctl.Response
+
+	np, err := s.renderNetworkPolicy(request)
+
+	if err != nil {
+		log.Error(err, "Could not render a NetworkPolicy from the incoming request")
+		return admissionctl.Errored(http.StatusBadRequest, err)
+	}
+
+	if hookconfig.IsPrivilegedNamespace(np.GetNamespace()) {
+		log.Info(fmt.Sprintf("%s operation detected on managed namespace: %s", request.Operation, np.GetNamespace()))
+		if isAllowedUser(request) {
+			ret = admissionctl.Allowed(fmt.Sprintf("User can do operations on NetworkPolicies"))
+			ret.UID = request.AdmissionRequest.UID
+			return ret
+		}
+		for _, group := range request.UserInfo.Groups {
+			if privilegedServiceAccountGroupsRe.Match([]byte(group)) {
+				ret = admissionctl.Allowed("Privileged service accounts do operations on NetworkPolicies")
+				ret.UID = request.AdmissionRequest.UID
+				return ret
+			}
+		}
+
+		ret = admissionctl.Denied(fmt.Sprintf("Prevented from accessing Red Mat managed resources. This is in an effort to prevent harmful actions that may cause unintended consequences or affect the stability of the cluster. If you have any questions about this, please reach out to Red Hat support at https://access.redhat.com/support"))
+		ret.UID = request.AdmissionRequest.UID
+		return ret
+	}
+
+	log.Info("Allowing access", "request", request.AdmissionRequest)
+	ret = admissionctl.Allowed("Non managed namespace")
+	ret.UID = request.AdmissionRequest.UID
+	return ret
+}
+
+// isclusterAdminUsers checks if the user or group is allowed to perform the action
+func isAllowedUser(request admissionctl.Request) bool {
+	if utils.SliceContains(request.UserInfo.Username, allowedUsers) {
+		return true
+	}
+
+	for _, group := range sreAdminGroups {
+		if utils.SliceContains(group, request.UserInfo.Groups) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (s *networkpoliciesruleWebhook) renderNetworkPolicy(req admissionctl.Request) (*networkPolicy, error) {
+	decoder, err := admissionctl.NewDecoder(&s.s)
+	if err != nil {
+		return nil, err
+	}
+	networkPolicy := &networkPolicy{}
+
+	if len(req.OldObject.Raw) > 0 {
+		err = decoder.DecodeRaw(req.OldObject, networkPolicy)
+	} else {
+		err = decoder.Decode(req, networkPolicy)
+	}
+	if err != nil {
+		return nil, err
+	}
+	return networkPolicy, nil
+}
+
+// GetURI implements Webhook interface
+func (s *networkpoliciesruleWebhook) GetURI() string {
+	return "/" + WebhookName
+}
```

Comment:
I think it's missing a validate function like this 

 `
func (s *networkpoliciesruleWebhook) Validate(request admissionctl.Request) bool {
	valid := true
	valid = valid && (request.UserInfo.Username != "")
	valid = valid && (request.Kind.Kind == "NetworkPolicy")

	return valid
}
`

Reply:
Good catch - added the function